$NetBSD: patch-am,v 1.3 2007/05/27 17:39:48 tonio Exp $

Use signed arithmetic in mutt_gecos_name to avoid an overflow.
From http://dev.mutt.org/hg/mutt/rev/47d08903b79b
Fix CVE-2007-2683

--- muttlib.c.orig	2007-04-03 19:52:56.000000000 +0200
+++ muttlib.c
@@ -540,7 +540,7 @@ char *mutt_gecos_name (char *dest, size_
     if (dest[idx] == '&')
     {
       memmove (&dest[idx + pwnl], &dest[idx + 1],
-	       MAX(destlen - idx - pwnl - 1, 0));
+	       MAX((ssize_t)(destlen - idx - pwnl - 1), 0));
       memcpy (&dest[idx], pw->pw_name, MIN(destlen - idx - 1, pwnl));
       dest[idx] = toupper ((unsigned char) dest[idx]);
     }
